lib/wickedbase.pm

```diff
@@ -560,26 +563,25 @@ sub post_run {
     $self->{wicked_post_run} = 1;
 
     $self->do_barrier('post_run');
-    my $result = eval {
-        $self->upload_wicked_logs('post');
-    };
-    record_info('Failed to upload post logs', $@, result => 'fail') unless defined $result;
+    $self->upload_wicked_logs('post');
 }
 
 sub pre_run_hook {
     my ($self) = @_;
+    if (get_var('VIRTIO_CONSOLE_NUM', 1) > 1) {
+        my $result = eval { add_serial_console('hvc1'); };
+        croak('Failed to add serial console : ' . $@) unless defined $result;
+    }
```

Comment:
What exception you are trying to catch here?

Reply:
you right - none , removed . and also we don't want to run this before **every** pre_run but just once

Reply:
I think `$self->select_serial_terminal()` makes sense for every run.